src/main/java/org/odpi/openmetadata/adapters/connectors/integration/eventschema/mapper/SchemaAttributeMapper.java

```diff
@@ -0,0 +1,238 @@
+            JsonObject typeObject = jsObject.getAsJsonObject(TYPE);
+            if (typeObject.has(FIELDS)) {
+                fields = typeObject.get(FIELDS);
+            }
+//            if (typeObject.has(TYPE) && typeObject.has(NAME)) {
+//                children.add(typeObject);
+//            }
+        }
+        if (fields != null && fields.isJsonArray()) {
+            var fieldsObject = (JsonArray) fields;
+            for (JsonElement field : fieldsObject) {
+                if (field.isJsonObject()) {
+                    children.add((JsonObject) field);
+                }
+            }
+        }
+
+        return children;
+    }
+
+    public void mapEgeriaSchemaAttribute() throws InvalidParameterException, PropertyServerException, UserNotAuthorizedException {
+        String parentQualifiedName = getParentQualifiedName();
+        schemaAttributeProperties.setDisplayName(getName());
+        schemaAttributeProperties.setDescription(getDoc());
+        schemaAttributeProperties.setTypeName("EventSchemaAttribute");
+        schemaAttributeProperties.setDataType(getType());
+        schemaAttributeProperties.setDefaultValue(getDefault());
+        schemaAttributeProperties.setIsNullable(isNullable());
+        schemaAttributeProperties.setQualifiedName(parentQualifiedName.concat(SEPARATOR).concat(getName()));
+    }
+
+    /**
+     * Get the qualifiedName of the parent SchemaType. This can either be a SchemaAttribute or an
+     *
+     * @return the qualifiedName of the parent or an empty String
+     */
+    private String getParentQualifiedName() throws InvalidParameterException, PropertyServerException, UserNotAuthorizedException {
+        EventTypeElement et;
+        SchemaAttributeElement attr;
+        if(Objects.equals(this.parentClass, EventTypeElement.class.getName())) {
+            et = context.getEventTypeByGUID(parentGUID);
+            return et.getProperties().getQualifiedName();
+        }
+        else {
+            attr = context.getSchemaAttributeByGUID(parentGUID);
+            return attr.getProperties().getQualifiedName();
+        }
+   }
+
+    public String createEgeriaSchemaAttribute() throws UnableToCreateSchemaAttributeException {
+        try {
+            List<SchemaAttributeElement> existingSchemaAttribute = context.getSchemaAttributesByName(schemaAttributeProperties.getQualifiedName(), "SchemaAttribute", 0, 0);
+            if(existingSchemaAttribute == null || existingSchemaAttribute.isEmpty() ) {
+                guid = context.createSchemaAttribute(parentGUID, schemaAttributeProperties);
+            } else {
+                guid = existingSchemaAttribute.get(0).getElementHeader().getGUID();
+                context.updateSchemaAttribute(guid, true, schemaAttributeProperties);
+            }
+        } catch (InvalidParameterException | UserNotAuthorizedException | PropertyServerException e) {
+            throw new UnableToCreateSchemaAttributeException();
```

Comment:
we should create a proper ffdc message and include the exception message

Reply:
I do this in the catch blocks of the calling class - the connector itself. This class does not have the audit log available.